### DISCUSSION TITLE
Is there a way to distingush Headers and Body in endpoints

### DISCUSSION THREAD

**Discussion by ashleshacc** [serverpod/serverpod#569] [Q&A]
@vlidholt @yahu1031 @II11II HI, Im new to backend development and server pod, wanted to ask is there a way to distinguish key:value which need to be added in a certain way so it part of headers and not body etc. currently Im able to get response in postman by adding all in body, any way to have header values and body separate so for a api call to give response, certain key:value need to be in body and some in headers?

**Comment by vlidholt** [comment]
Can you describe the use case for why you want to do this? It's somewhat tricky to create correct serverpod posts through postman, but it's very easy from your Flutter code.

**Reply by ashleshacc** [reply]
my scenario is so for example get user detail api, i can have specific fields like , device details, access token etc in headers, and just the requried thing like user id in json body, so any way i can separate these, so that if i create a postman collection or something, i can have it added like that.

**Reply by vlidholt** [reply]
With Serverpod you don't really have to worry about JSON or your headers. Just create a serializable object through a yaml-file and you can return that object from one of your calls. It may be a smart starting point to check out one of the tutorials to get started. :)

https://docs.serverpod.dev/tutorials

**Reply by ashleshacc** [reply]
ok thank you, but just to access apis from postman, just for testing purposes, is it possible to have it so that i have to define certain headers n all, or not possible currently? only way to test in postman to add all in json Body?

**Reply by vlidholt** [reply]
It's definitely possible to test the APIs this way. However, the documentation is a bit sparse on exactly what the protocol looks like and what is required to make the requests. This is an area we will improve in the future.
